### PR TITLE
[libsndfile] Update to 1.1.0

### PR DESCRIPTION
**Pull request by evpobr**
* Update to 1.1.0
* Add `mpeg` and `regtest` features to sync with upstream.

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@evpobr ,Thanks for your pr, Looks the ci tested failed on many triplet,
This is error log, could you please take a look:
https://dev.azure.com/vcpkg/public/_build/results?buildId=69470&view=artifacts&pathAsName=false&type=publishedArtifacts

**Comment by evpobr** [comment]
@JonLiu1993 , i guess it is bug in upstream, fixing.

**Comment by evpobr** [comment]
@JonLiu1993 , fixed.

File: ports/libsndfile/vcpkg.json [hunk 1]
```diff
@@ -1,20 +1,50 @@
 {
   "name": "libsndfile",
-  "version-string": "1.0.31",
-  "description": "Library to read, write and manipulate many soundfile types. Authored by Eric de Castro Lopo",
+  "version-semver": "1.1.0",
```
Comment (JonLiu1993) [inline ports/libsndfile/vcpkg.json:3]: @evpobr, Could you please confirm whether the library officially adheres to semantic versioning or not (breaking changes -> bump major version, non-breaking changes -> bump minor version, non-breaking bug fixes -> bump patch version)?

Comment (Neumann-A) [reply ports/libsndfile/vcpkg.json:3]: doesn't matter `version-semver` shouldn't be used. Use version instead.
```suggestion
  "version": "1.1.0",
```

Comment (evpobr) [reply ports/libsndfile/vcpkg.json:3]: I didn't understand. so should I or not use `version-semver`?

https://vcpkg.readthedocs.io/en/latest/specifications/versioning/#2-specifying-package-versions

It says here that it is possible.

Comment (evpobr) [reply ports/libsndfile/vcpkg.json:3]: And yes, we use SemVer:

https://github.com/libsndfile/libsndfile/blob/master/CHANGELOG.md

Comment (JonLiu1993) [reply ports/libsndfile/vcpkg.json:3]: ''The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html)." Thanks for your clarification

Comment (JonLiu1993) [reply ports/libsndfile/vcpkg.json:3]: > I didn't understand. so should I or not use `version-semver`?
> 
> https://vcpkg.readthedocs.io/en/latest/specifications/versioning/#2-specifying-package-versions
> 
> It says here that it is possible.

We are cautious about using "version-semver", if Upstream explicitly says adheres to Semantic Versioning we allow to use “version-semver”

**Comment by JonLiu1993** [comment]
All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

**Comment by strega-nil-ms** [comment]
Thanks so much!

**Pull request merged**